File: solutions/c/easy/bitwise-operators-in-c/solution-approach1.c
// ──────────────────────────────────────────────────
// Problem     Bitwise Operators
// Difficulty  Easy
// Subdomain   Conditionals and Loops
// Platform    HackerRank
// Language    c
// Status      Accepted
// Submitted   2026-06-28, 08:28 p.m.
// ──────────────────────────────────────────────────

#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

void calculate_the_maximum(int n, int k)
{
    int max_and = 0;
    int max_or = 0;
    int max_xor = 0;

    for(int a = 1; a <= n; a++)
    {
        for(int b = a + 1; b <= n; b++)
        {
            int and = a & b;
            int or = a | b;
            int xor = a ^ b;

            if(and < k && and > max_and)
            {
                max_and = and;
            }

            if(or < k && or > max_or)
            {
                max_or = or;
            }

            if(xor < k && xor > max_xor)
            {
                max_xor = xor;
            }
        }
    }

    printf("%d\n", max_and);
    printf("%d\n", max_or);
    printf("%d\n", max_xor);
}

int main()
{
    int n, k;

    scanf("%d %d", &n, &k);

    calculate_the_maximum(n, k);

    return 0;
}

File: solutions/python3/easy/nested-list/solution-approach1.py
# ──────────────────────────────────────────────────
# Problem     Nested Lists
# Difficulty  Easy
# Subdomain   Basic Data Types
# Platform    HackerRank
# Language    python3
# Status      Accepted
# Submitted   2026-06-30, 10:43 p.m.
# ──────────────────────────────────────────────────

if __name__ == '__main__':
    # Initialize the list correctly inside the block
    students = []
    
    # Read total number of students
    for _ in range(int(input())):
        name = input()
        score = float(input())
        students.append([name, score])
        
    # Extract unique scores, sort them, and select the second element
    unique_scores = sorted(list(set([student[1] for student in students])))
    second_lowest = unique_scores[1]
    
    # Filter names matching the second lowest score
    second_lowest_students = [student[0] for student in students if student[1] == second_lowest]
    
    # Sort names alphabetically and print
    for name in sorted(second_lowest_students):
        print(name)
